### PR TITLE
docs(rbac): add documentation for api and known permissions

### LINKED ISSUE #978: [RBAC plugin] Documentation updates
### What do you want to improve?

We should explore enhancing the documentation to incorporate more user-relevant information.

### What does the current documentation state?

Presently, our documentation covers only the fundamental steps of configuring the RBAC backend plugin. Additionally, there is an issue with the documentation concerning RHDH.

### What would you like to be added/updated in the documentation?

- Remove the note stating that RHDH users enjoy the benefit of RBAC being enabled by default.
  - Currently, users are responsible for configuring RBAC, including enabling it.

- Add documentation for the API specs

- Add documentation for some of the known permissions that are available in RHDH
  - Kubernetes
  - Catalog
  - RBAC
  - Scaffolder
  - Jenkins

### PR DESCRIPTION
**Pull request by PatAKnight**
## Description

Adds and edits documentation for the RBAC Backend plugin. Included is information on various API endpoints, some of the known permissions that are included as dynamic plugins within RHDH / Showcase, and some minor updates addressing documentation issues.

## Fixes

- Fixes #978 

## How to test changes / Special notes to the reviewer

None

### CODE REVIEW

File: plugins/rbac-backend/docs/apis.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+PUT </api/permission/roles/:kind/:namespace/:name>
+ex. <http://localhost:7007/api/permission/roles/role/default/test_admin>
+
+Updates a specified role.
+
+Request Parameters:
+
+| Parameter name | Description           | Type   |
+| -------------- | --------------------- | ------ |
+| kind           | Kind of the role      | String |
```
Comment (invincibleJai) [inline plugins/rbac-backend/docs/apis.md:259]: just for my understanding, what does kind of role mean here? or possible values?

will it be `role`, `user` , `group`

File: plugins/rbac-backend/docs/apis.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+  {
+    "entityReference": "role:default/test",
+    "permission": "catalog-entity",
+    "policy": "read",
+    "effect": "allow"
+  },
+  {
+    "entityReference": "role:default/test",
+    "permission": "catalog.entity.create",
+    "policy": "use",
```
[the remark this answers is not in the document]
Comment (PatAKnight) [reply plugins/rbac-backend/docs/apis.md]: Fixed to now say `create`

File: plugins/rbac-backend/docs/apis.md (hunk 1 above)
Comment (PatAKnight) [reply plugins/rbac-backend/docs/apis.md]: Updated it to now have just `role` to hopefully reduce confusion.

**Review by AndrienkoAleksandr — APPROVED** [review]

**Comment by PatAKnight** [comment]
Updated the order as requested by @AndrienkoAleksandr

**Pull request merged**